### PR TITLE
[Doc] API Doc update for mxnet and tf, remove some degree check

### PR DESCRIPTION
**Pull request by sneakerkg**
## Description
1.API Doc update for mxnet and tf.
2.Remove degree check for nnconv and ginconv.
3.GAT has two fc layers for bipartite.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/nn/mxnet/conv/gatconv.py [hunk 1]
```diff
@@ -53,16 +146,26 @@ def __init__(self,
...
         self._in_src_feats, self._in_dst_feats = expand_as_pair(in_feats)
         self._in_feats = in_feats
         self._out_feats = out_feats
+        self._allow_zero_in_degree = allow_zero_in_degree
         with self.name_scope():
-            self.fc = nn.Dense(out_feats * num_heads, use_bias=False,
-                               weight_initializer=mx.init.Xavier(magnitude=math.sqrt(2.0)),
-                               in_units=in_feats)
+            if isinstance(in_feats, tuple):
+                self.fc_src = nn.Dense(out_feats * num_heads, use_bias=False,
```
Comment (yzh119) [inline python/dgl/nn/mxnet/conv/gatconv.py:159]: If we change the behavior to two separate linear projection layers.
The `test_nn` and clusterGAT example have to be refactored accordingly.

File: python/dgl/nn/mxnet/conv/gatconv.py [hunk 2]
```diff
@@ -84,33 +187,69 @@ def __init__(self,
...
+                                   'calling `g = dgl.add_self_loop(g)` will resolve '
+                                   'the issue. Setting ``allow_zero_in_degree`` '
+                                   'to be `True` when constructing this module will '
+                                   'suppress the check and let the code run.')
+
             if isinstance(feat, tuple):
                 h_src = self.feat_drop(feat[0])
                 h_dst = self.feat_drop(feat[1])
-                feat_src = self.fc(h_src).reshape(
+                feat_src = self.fc_src(h_src).reshape(
```
Comment (yzh119) [inline python/dgl/nn/mxnet/conv/gatconv.py:250]: Considering there are cases we apply GAT-on-homograph to bipartite graph, I suggest add a if-statement here:
```suggestion
                if not has_attr(self, fc_src): self.fc_src, self.fc_dst = self.fc, self.fc
                feat_src = self.fc_src(h_src).reshape(
```

**Comment by yzh119** [comment]
GAT test: https://github.com/dmlc/dgl/blob/master/tests/pytorch/test_nn.py#L385-L392
ClusterGAT example: https://github.com/dmlc/dgl/blob/master/examples/pytorch/ogb/cluster-gat/main.py (note the difference between forward and inference function).

File: python/dgl/nn/mxnet/conv/gatconv.py (hunk 2 above)
Comment (sneakerkg) [reply python/dgl/nn/mxnet/conv/gatconv.py:250]: When will we apply a GAT-on-homo to bipartite graph? As discussed before, for sample-based inference, we will just is_block() and  graph.number_of_dst_nodes() to get the dst nodes, no need to make the input tuple.

**Comment by sneakerkg** [comment]
> GAT test: https://github.com/dmlc/dgl/blob/master/tests/pytorch/test_nn.py#L385-L392
> ClusterGAT example: https://github.com/dmlc/dgl/blob/master/examples/pytorch/ogb/cluster-gat/main.py (note the difference between forward and inference function).

Fixed ClusterGAT by sending a tensor not tuple when g is a block. 

For the test case, if no fc_src, we will go for fc. Still not quite sure whether we should provide this flexibility.

**Pull request merged**